### PR TITLE
some update of Remote-Playback-API

### PR DESCRIPTION
**Pull request by skyclouds2001**
### Description

This is part of the #22441 

This also inherits from #26541 which has been closed without merged

---

add Remote Playback API to api group data

update `{{APIRef("Remote Playback API")}}` in `RemotePlayback` interface and its members, also `HTMLMediaElement.disableRemotePlayback`

initial `HTMLMediaElement.remote` documentation

add *Extensions to other interfaces* section to api summary

remove `{{experimental_inline}}` macros of `disableremoteplayback` property of `audio` or `video` element

some other style pretty updates

### Additional details

working on https://developer.mozilla.org/en-US/docs/Web/API/Remote_Playback_API
reference to https://w3c.github.io/remote-playback/

### CODE REVIEW

File: files/en-us/web/api/htmlmediaelement/remote/index.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+title: "HTMLMediaElement: remote property"
+short-title: remote
+slug: Web/API/HTMLMediaElement/remote
+page-type: web-api-instance-property
+browser-compat: api.HTMLMediaElement.remote
+---
+
+{{APIRef("Remote Playback API")}}
+
+The **`remote`** read-only property of the {{domxref("HTMLMediaElement")}} interface return a {{domxref("RemotePlayback")}} object instance associated with the media element. This is the entry point of the {{domxref("Remote Playback API", "Remote Playback API", "", "nocode")}}.
```
Comment (skyclouds2001) [inline files/en-us/web/api/htmlmediaelement/remote/index.md:11]:
```suggestion
The **`remote`** read-only property of the {{domxref("HTMLMediaElement")}} interface return a {{domxref("RemotePlayback")}} object instance associated with the media element. This is the entry point of the {{domxref("Remote Playback API", "", "", "nocode")}}.
```

File: files/en-us/web/api/remote_playback_api/index.md [hunk 2]
```diff
@@ -24,6 +25,13 @@ The API enables a page, which has an media element such as a video or audio file
 - {{domxref("RemotePlayback")}}
   - : Allows the page to detect availability of remote playback devices, then connect to and control playing on these devices.
 
+## Extensions to other interfaces
```
Comment (skyclouds2001) [inline files/en-us/web/api/remote_playback_api/index.md]:
```suggestion
### Extensions to other interfaces
```

File: files/en-us/web/api/htmlmediaelement/remote/index.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+title: "HTMLMediaElement: remote property"
+short-title: remote
+slug: Web/API/HTMLMediaElement/remote
+page-type: web-api-instance-property
+browser-compat: api.HTMLMediaElement.remote
+---
+
+{{APIRef("Remote Playback API")}}
+
+The **`remote`** read-only property of the {{domxref("HTMLMediaElement")}} interface return a {{domxref("RemotePlayback")}} object instance associated with the media element. This is the entry point of the {{domxref("Remote Playback API", "", "", "nocode")}}.
```
Comment (teoli2003) [inline files/en-us/web/api/htmlmediaelement/remote/index.md:11]: _A_ or _the_ ? If we call this property twice, do we get the same `RemotePlayback` object or a different (new) one?

```suggestion
The **`remote`** read-only property of the {{domxref("HTMLMediaElement")}} interface returns the {{domxref("RemotePlayback")}} object associated with the media element. The `RemotePlayback` object allow the control of remote devices playing the media.
```

I don't think that saying that something is an entry point of an API is useful. The division in APIs is arbitrary and not really useful to developpers. They will click on `RemotePlayback` and from them can get an overview, if they want.

File: files/en-us/web/api/htmlmediaelement/remote/index.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+## Value
+
+A {{domxref("RemotePlayback")}} object instance.
```
Comment (teoli2003) [inline files/en-us/web/api/htmlmediaelement/remote/index.md:15]: See above. I assumed that it always the same object.
```suggestion
The associated {{domxref("RemotePlayback")}} object.
```

File: files/en-us/web/api/htmlmediaelement/remote/index.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+## Example
+
+```js
+const el = document.createElement("audio");
+const remotePlayback = el.remote;
+
+remotePlayback.watchAvailability((availability) => {
+  // do something when the availability changes
```
Comment (teoli2003) [inline files/en-us/web/api/htmlmediaelement/remote/index.md:24]:
```suggestion
  // Do something when the availability changes
```

File: files/en-us/web/api/remote_playback_api/index.md [hunk 6]
```diff
@@ -3,6 +3,7 @@ title: Remote Playback API
 slug: Web/API/Remote_Playback_API
 page-type: web-api-overview
 browser-compat: api.RemotePlayback
+spec-urls: https://w3c.github.io/remote-playback/
```
Comment (teoli2003) [inline files/en-us/web/api/remote_playback_api/index.md:6]: I don't like the sentence below here, but GitHub doesn't let me propose a change.

Can you change:
_The Remote Playback API extends the {{domxref("HTMLMediaElement")}} to enable the control of remote playback of media from a webpage._

to something like
_The Remote Playback API extends the {{domxref("HTMLMediaElement")}} to enable the control of media played on a remote device._

I didn't understand the original sentence.

**Review by teoli2003 — CHANGES_REQUESTED** [review]
A few changes. Even if I need to perform a quick second pass, I think this is close to be merged.

**Review by teoli2003 — APPROVED** [review]
LGTM!

I have double-checked if remote always returns the same object, and it does-

The webidl is very clear:

`SameObject readonly attribute RemotePlayback remote;`

**Pull request merged**